src/neo/Ledger/Blockchain.cs

```diff
@@ -237,9 +273,34 @@ private VerifyResult OnNewBlock(Block block)
                 // We can store the new block in block_cache and tell the new height to other nodes after Persist().
                 system.LocalNode.Tell(Message.Create(MessageCommand.Ping, PingPayload.Create(Singleton.Height)));
             }
+            else
+            {
+                if (block.Index + 99 >= HeaderHeight)
+                    system.LocalNode.Tell(new LocalNode.RelayDirectly { Inventory = block });
+                if (block.Index == HeaderHeight + 1)
+                {
+                    Header header = block.Header;
+                    recentHeaders.Add(header);
```

Comment:
Why add the persisted block to the headers cache?

Reply:
If the block is 1 higher than highest index in header cache, we can treat this as a valid header msg.